WordPress/pattern-directory
# Theme: Try a `PatternGrid` component to display patterns

See #62 — This adds a few components to create a dynamic pattern grid. It uses the same structure and CSS introduced in #68, but also adds an iframe'd pattern preview.

This mainly gets the components in place, but doesn't touch anything data-y yet, not really. 

I'm using `select( coreStore ).getEntityRecords( 'postType', 'wporg-pattern' );` to get the list of patterns, but this generates an API request that only works for logged in editors. We'll want to create our own store to fetch just the plain `wporg-patterns` endpoint, not the edit context. See some discussion on https://github.com/WordPress/pattern-directory/issues/66#issuecomment-812450704.

❓ Would we consider this OK-enough to merge, or should we try to get the data side in first?

⚠️  I know the previews aren't great (design-wise), but I think I'd rather iterate on those in future PRs. Open to debate on that too 🙂 

### Screenshots

<img width="1045" alt="Screen Shot 2021-04-02 at 6 18 20 PM" src="https://user-images.githubusercontent.com/541093/113458342-00c80d00-93e0-11eb-81ef-873d23fcc0d1.png">

### How to test the changes in this Pull Request:

0. Be logged in (see above re: API)
1. Load any archive page
2. It should render the list of patterns _(It will always show you the latest patterns, no filtering yet)_
3. Click on a pattern
4. It should take you to the pattern detail view, no change here
